actions/views/root.js

```diff
@@ -15,7 +17,7 @@ export function loadMeAndConfig() {
             dispatch(getLicenseConfig()),
         ];
 
-        if (document.cookie.indexOf('MMUSERID=') > -1) {
+        if (document.cookie.indexOf('MMUSERID=') > -1 && LocalStorageStore.getWasLoggedIn()) {
```

Comment:
This might not be the cause of the ticket and i don't see how this will help solve the issue here? 

We are adding another check to make the call for getting teamLlist here. which will not help as the reason why we see `select_team` page in the first place is because we don't have teams 

For me this seems like this happens because after login submit we are redirecting user to defaultTeam by the function `redirectUserToDefaultTeam` without ensuring we have `teamsList`.

Reply:
It happens during logging in since after logout, it's adding redirect_to query parameter on "/login" path.  I maybe wrong but I'm pretty sure about this fix.

Reply:
> it's adding redirect_to query parameter on "/login" path

Yes but does it cause any issues wrt to `select_team` screen? 

I don't see how this will fix the issue even if that is the reason as we are not calling the `loadMe` now teams now and it will cause bigger problems from my guess?

Re: https://github.com/mattermost/mattermost-webapp/blob/master/components/root/root.jsx#L229  this should not hit this path as we are checking for `/` as this will have redirect params from your inference.

i remember Dylan saying this happening when logging in with an automated tool so it wasn't the path as you were referring here.

@DHaussermann Can you check if you were able to replicate this again with the tool on CI when you find time?

Reply:
I am not able to find any issue with the `select_team` screen.

You're right that we expect not to hit that path but for a reason (on re-render) it's being hit since  the `MMUSERID` is still present in the cookie for some time after the user logged out, and so with that additional flag, we'll only call `loadMe` if the user is actually logged in.

Yup it's captured on automated tool, since after logout it will do login for the next test.

To easily reproduce the issue, hit refresh after the user logged in.  After that, do the logout.  You'll notice that the `select_team` is added to `redirect_to`.  I have screen recording in case you'd like to see in action - https://www.youtube.com/watch?v=Qh5anqzGp74

Reply:
Are we maybe not clearing MMUSERID at the right time? Alternatively, maybe we could change how the Root component redirects to `/select_team` since it's looking at the return value of the `loadMe` from this and that always responds successful

Reply:
@saturninoabril Aah. Interesting nice debugging 👍 

I don't know how to go about it now on this. If any of suggestions from HH seems to provide a non trivial fix then sure go ahead otherwise I am ok with this as ABC is looking at changing the logic around it in the coming release.

Reply:
I've tried several attempts today including all your suggestions but didn't work for me.  I think there are several things happening here like re-render, mount/unmount, etc. that makes it harder to trace the root cause.  I'm convince that the most sensible/practical to do at least for now is my approach.
I suggest to do other things separately if there will be similar issue in the future.

No change in this PR; should be ready for dev review.

Reply:
Alright. While I'm not a fan of this fix, we've probably spent enough time looking at it. That whole log in/log out flow is kind of a mess anyway

Reply:
FYI this fix was reverted as it was causing issues with SAML login

Reply:
Well noted.  I'l' reopen the ticket - MM-12897.